src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const grouped: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const itemKey = String(item[key]);
+
+    if (!Object.prototype.hasOwnProperty.call(grouped, itemKey)) {
```

Comment:
I think it would simpler but your solution also good👍
```suggestion
    if (!grouped[itemKey]) {
```